modules/c++/io/tests/serializeTest1.cpp

```diff
@@ -73,7 +73,9 @@ class A : public Serializable
     {
 	std::string toFill;
 	sys::byte b[1];
-	while (true)
+	// Work-around warning C4127: conditional expression is constant;
+	// see https://msdn.microsoft.com/en-us/library/6t66728h.aspx and http://stackoverflow.com/questions/1946445/c-c-how-to-use-the-do-while0-construct-without-compiler-warnings-like-c412
+	for (;;) // while (true)
```

Comment:
Yeah, it is a tradeoff.  The other thing is that a good chunk of our compilation is on Linux (where warnings are much more sparse because otherwise I would go crazy) and even with `-Wall` on gcc doesn't emit this one, so arguably we've been getting by fine without it.  I wish there were more fine-grained warnings you could turn on/off such that you could tell the compiler that the really obvious stuff like `while (true)` was done with intention and so doesn't need to be a warning, though the compiler would have to be pretty smart to distinguish that I suppose.

To me, the possibility of someday making a logic error that leads to dead code is worth the risk to avoid needing to explain to every new hire that they need to write `for (;;)`, so I'd vote to add this to build.py.